FontNode.js

```diff
@@ -0,0 +1,50 @@
+import path from 'path';
+import './lib/inflate.js';
+import './lib/unbrotli.js';
+import './Font.js';
+import fs from 'fs';
+import { getFontCSSFormat } from './Font.js';
+
+fs.readFileAsync = (filename) => {
+    return new Promise((resolve, reject) => {
+        fs.readFile(filename, (err, data) => {
+            if (err) reject(err);
+            else resolve(data);
+        });
+    });
+};
+
+export default class FontNode extends Font {
+    constructor(fileRaw, options = {}) {
+        if (fileRaw) {
+            const file = path.normalize(path.resolve(fileRaw)) || '';
+            const name = file.replace(/^.*\/(.*?)(?:\..*?)$/, '$1');
+            const font = super(name, options);
+            font.src = file;
+            return new Promise(function (resolve, reject) {
+                font.onload = data => {
+                    resolve(data.detail.font);
+                };
+                font.onerror = error => {
+                    reject(error);
+                };
+            });
+        }
+    }
+
+    set src(url) {
+        this.__src = url;
+        this.loadFont(url);
+    }
+
+    loadFont = (url) => {
```

Comment:
Any reasons this is renamed to loadFile in the rewrite?

Reply:
Mostly the fact that it doesn't load a font: it loads a file, and hopefully that's  font but it might not be. The information that we're assuming it's a font is already captured by the fact that the class is `Font`.